overops-functions/src/main/java/com/takipi/udf/microsoftteams/MicrosoftTeamsAnomalyFunction.java

```diff
@@ -0,0 +1,157 @@
+package com.takipi.udf.microsoftteams;
+
+import com.google.common.base.Strings;
+import com.google.gson.Gson;
+import com.takipi.api.client.ApiClient;
+import com.takipi.api.client.data.view.SummarizedView;
+import com.takipi.api.client.request.event.EventsVolumeRequest;
+import com.takipi.api.client.request.view.ViewRequest;
+import com.takipi.api.client.result.event.EventsResult;
+import com.takipi.api.client.result.view.ViewResult;
+import com.takipi.api.client.util.validation.ValidationUtil;
+import com.takipi.api.core.url.UrlClient;
+import com.takipi.udf.ContextArgs;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+import static com.takipi.udf.microsoftteams.MicrosoftTeamsChannelFunction.getContextArgs;
+import static com.takipi.udf.microsoftteams.MicrosoftTeamsChannelFunction.getEnvironmentName;
+import static com.takipi.udf.microsoftteams.MicrosoftTeamsUtil.*;
+
+public class MicrosoftTeamsAnomalyFunction {
+
+    public static String validateInput(String rawInput) {
+        return getInput(rawInput).toString();
+    }
+
+    public static void execute(String rawContextArgs, String rawInput) {
+        try {
+            executeImplementation(rawContextArgs, rawInput);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void executeImplementation(String rawContextArgs, String rawInput) {
+        logUDFInput(rawContextArgs, rawInput);
+        MicrosoftTeamsAnomalyInput input = getInput(rawInput);
+        ContextArgs args = (new Gson()).fromJson(rawContextArgs, ContextArgs.class);
+        if (!args.validate())
```

Comment:
Please have `{}` surrounding if code blocks, even if it's just a single line. Also, this should probably check `viewValidate`
In this case:

```
if (!args.validate()) {
    throw new IllegalArgumentException("Bad context args: " + rawContextArgs);
}
if (!args.viewValidate()) {
    return;
}
```